### PR TITLE
Refactor Workers

### PR DESCRIPTION
**Pull request by andreespirela**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,7 +10,8 @@ members = [
   "./crates/flaky_test",
   "./crates/sb_os",
   "./crates/node",
-  "./crates/cpu_timer"
+  "./crates/cpu_timer",
+  "./crates/event_manager"
```
Comment (laktek) [inline Cargo.toml:14]: can we rename this crate to `event_worker`?

File: crates/base/src/rt_worker/worker_ctx.rs [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+pub async fn create_worker(
+    init_opts: WorkerContextInitOpts,
+) -> Result<mpsc::UnboundedSender<WorkerRequestMsg>, Error> {
+    let (worker_boot_result_tx, worker_boot_result_rx) = oneshot::channel::<Result<(), Error>>();
+    let (unix_stream_tx, unix_stream_rx) = mpsc::unbounded_channel::<UnixStream>();
+    let worker_init = Worker::new(&init_opts)?;
+
+    let worker: Box<dyn WorkerHandler> = Box::new(worker_init);
+
+    // Downcast to call the method in MyStruct
```
Comment (laktek) [inline crates/base/src/rt_worker/worker_ctx.rs:158]: what is MyStruct?

File: crates/base/src/rt_worker/worker_ctx.rs [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+) -> Result<mpsc::UnboundedSender<UserWorkerMsgs>, Error> {
+    let (user_worker_msgs_tx, mut user_worker_msgs_rx) =
+        mpsc::unbounded_channel::<UserWorkerMsgs>();
+
+    let user_worker_msgs_tx_clone = user_worker_msgs_tx.clone();
+
+    let _handle: tokio::task::JoinHandle<Result<(), Error>> = tokio::spawn(async move {
+        let mut worker_pool = WorkerPool::new(worker_event_sender, user_worker_msgs_tx_clone);
+
+        loop {
```
Comment (laktek) [inline crates/base/src/rt_worker/worker_ctx.rs:256]: Nice work on simplifying the worker pool implementation.

File: crates/base/src/rt_worker/worker_pool.rs [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+use tokio::sync::oneshot::Sender;
+
+pub struct WorkerPool {
+    pub worker_event_sender: Option<mpsc::UnboundedSender<WorkerEventWithMetadata>>,
+    pub user_workers: HashMap<u64, UserWorkerProfile>,
+    pub user_worker_msg_tx: mpsc::UnboundedSender<UserWorkerMsgs>,
+}
+
+pub enum CreationCodes {
+    AlreadyExistent,
```
Comment (laktek) [inline crates/base/src/rt_worker/worker_pool.rs:25]: rename this to `AlreadyExist`?

File: crates/event_manager/js_interceptors.rs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+use crate::events::{EventMetadata, LogEvent, LogLevel, WorkerEvents};
+use crate::WorkerEventWithMetadata;
+use deno_core::error::AnyError;
+use deno_core::op;
+use deno_core::OpState;
+use log::error;
+use tokio::sync::mpsc;
+
+#[op]
+pub fn op_user_worker_log(state: &mut OpState, msg: &str, is_err: bool) -> Result<(), AnyError> {
```
Comment (laktek) [inline crates/event_manager/js_interceptors.rs:10]: Cool! thanks for moving this op to events crate.

**Review by laktek — APPROVED** [review]
LGTM! minor suggestions on naming.

File: crates/base/src/rt_worker/worker_ctx.rs (hunk 2 above)
Comment (andreespirela) [reply crates/base/src/rt_worker/worker_ctx.rs:158]: Lol. Will clean this

File: Cargo.toml (hunk 1 above)
Comment (andreespirela) [reply Cargo.toml:14]: Awesome

File: crates/base/src/rt_worker/worker_pool.rs (hunk 4 above)
Comment (andreespirela) [reply crates/base/src/rt_worker/worker_pool.rs:25]: Will do

**Pull request merged**